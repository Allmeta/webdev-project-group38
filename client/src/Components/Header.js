import React, { Component } from 'react'
import styled from 'styled-components'
import Logo from '../images/filmguru-logo2.jpg'
import SearchForm from './SearchForm.js'

// The logo uses this hex-color: #4553F4

const HDiv = styled.div`
  position:relative;
  padding: 0;
  padding-left:10%;
  padding-right:10%;
  margin: 0;
  background-color: #4553F4;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  

  @media screen and (max-width: 500px) {
    flex-direction: column;
  }
`
const ImgStyle = styled.img`
  width:200px;
  float: left;
`
const SearchFields = styled.div`
  float:left;
  position:absolute;
  width:1000px;
  left:40%;
  top:30px;
`

class Header extends Component {
  render () {
    return (
      <HDiv>
        <ImgStyle src={Logo} />
        <SearchFields>
          <SearchForm />
        </SearchFields>
      </HDiv>
    )
  }
}
export default Header
