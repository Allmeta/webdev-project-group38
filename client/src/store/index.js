import { createStore, applyMiddleware, compose } from 'redux'
import { SearchFormReducer } from "../reducers/SearchFormReducer";
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(SearchFormReducer, composeEnhancers(
  applyMiddleware(thunk)
));

export default store;