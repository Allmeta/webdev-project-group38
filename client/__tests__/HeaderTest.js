import React from 'react'
import Header from '../src/Components/Header.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import store from '../src/store/index.js'
import { Provider } from 'react-redux'

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Header /></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})
