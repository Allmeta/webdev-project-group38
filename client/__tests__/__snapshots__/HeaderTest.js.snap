// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c4 {
  right: 50;
}

.c0 {
  position: relative;
  padding: 0;
  padding-left: 10%;
  padding-right: 10%;
  margin: 0;
  background-color: #4553F4;
  font-family: Verdana,Geneva,Tahoma,sans-serif;
  color: white;
  text-align: center;
  height: 110px;
}

.c1 {
  width: 200px;
  float: left;
}

.c2 {
  float: left;
  position: absolute;
  width: 1000px;
  left: 40%;
  top: 50px;
}

.c3 {
  float: right;
  position: absolute;
  width: 1000px;
  left: 60%;
  top: 50px;
}

@media screen and (max-width:500px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<header
  className="ReactStickyHeader_root"
>
  <div
    className="ReactStickyHeader_fixed"
  >
    <div
      className="Header_root"
    >
      <div
        className="c0"
      >
        <img
          className="c1"
          src="test-file-stub"
        />
        <div
          className="c2"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="fields"
            >
              <div
                className="three wide field"
              >
                <div
                  className="ui input"
                >
                  <input
                    name="title"
                    onChange={[Function]}
                    placeholder="Title search"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="two wide field"
              >
                <button
                  className="ui icon button"
                  onClick={[Function]}
                  role="button"
                  type="submit"
                >
                  <i
                    aria-hidden="true"
                    className="search icon"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
        <div
          className="c3"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="c4"
            >
              <div
                className="fields"
              >
                <div
                  className="three wide field"
                >
                  <div
                    className="ui input"
                  >
                    <input
                      name="filterQuery"
                      onChange={[Function]}
                      placeholder="Filter on title..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <button
                  className="ui grey icon right floated right labeled button"
                  onClick={[Function]}
                  role="button"
                >
                  Rating
                  <i
                    aria-hidden="true"
                    className=" sort numeric up big icon"
                  />
                </button>
                <div
                  className="c4"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ReactStickyHeader_midground"
    style={
      Object {
        "backgroundColor": undefined,
        "backgroundImage": undefined,
        "height": null,
        "top": null,
      }
    }
  />
  <div
    className="ReactStickyHeader_background-bg"
    style={
      Object {
        "backgroundColor": undefined,
        "backgroundImage": undefined,
      }
    }
  />
  <div
    className="ReactStickyHeader_foreground"
    style={
      Object {
        "backgroundColor": undefined,
        "backgroundImage": undefined,
        "opacity": 1,
      }
    }
  />
  <section />
</header>
`;
